packages/marshal/src/passStyleOf.js

```diff
@@ -190,6 +172,50 @@ function isPassByCopyRecord(val) {
   return true;
 }
 
+// Below we have a series of predicate functions and their (curried) assertion
+// functions. The semantics of the assertion function is just to assert that
+// the corresponding predicate function would have returned true. But it
+// reproduces the internal tests so failures can give a better error message.
+
+/**
+ * @callback CheckFn
+ * @param {any} cond
+ * @param {any} details
+ */
```

Comment:
Is it possible to declare the return type such that truthy tells typescript the condition is truthy? That would happen both for `assertTruthy` and for your default identity function.

The `details` parameter of `assert` is optional. I don't see any reason to require it here. Also, may as well preserve `assert`'s `Details` type here.

```suggestion
 * @callback CheckFn
 * @param {any} cond
 * @param {Details=} details
 */
```

Reply:
> Is it possible to declare the return type such that truthy tells typescript the condition is truth?

Unfortunately, that's a misdesign in Typescript's `asserts X` return type.  It can only return `undefined`.  And you can't do anything more specific than a static type (not an argument's actual value) if you return something else.  Dependent typing it is not.

I would report this to Typescript, if I could figure out how to describe the problem well.

> The `details` parameter of `assert` is optional. I don't see any reason to require it here. Also, may as well preserve `assert`'s `Details` type here.

Done.